### ISSUE TITLE
Thread safety of ParseMessages and SendMsg

### ISSUE THREAD

**Issue by orca-hydromancer** [ttlappalainen/NMEA2000#282]
Can `NMEA2000::ParseMessages` and `NMEA2000::SendMsg` be called from different threads safely?

**Comment by ttlappalainen** [comment]
Only by protecting with e.g., mutex that you do not call them at same time.

**Comment by david-alonso-orca** [comment]
Which are the shared resources which need to be protected? Is there any shared buffer?

**Comment by ttlappalainen** [comment]
Yes. ParseMessages may also do SendMsg when responding to internal requests. ParseMessages also does callbacks, if you have defined them. Depending of solution callback may also generate SendMsg. Both are fast routines. I use global NMEA2000 mutex and lock calls with that.
